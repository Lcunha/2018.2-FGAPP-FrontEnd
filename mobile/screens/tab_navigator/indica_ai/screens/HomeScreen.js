import React, { Component } from "react";
import {
  Text
} from "react-native";
import { connect } from 'react-redux';
import { Provider } from 'react-redux'
import { authAction } from '../actions'
import { bindActionCreators } from 'redux';

class HomeScreen extends Component {

  constructor(props){
    super();
    props.authAction(props.user_token)
  }

  render() {

    return (
        <Text>{this.props.user_token}</Text>
    );
  }
}

const mapStateToProps = store => ({})

const mapDispatchToProps = dispatch => (
    bindActionCreators({ authAction }, dispatch))

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
