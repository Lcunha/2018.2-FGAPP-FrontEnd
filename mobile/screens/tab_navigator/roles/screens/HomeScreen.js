import React, { Component } from "react";
import { 
    View,
    Text,
    StyleSheet,
    Button,
    TouchableOpacity,
    Alert
} from "react-native";
import cadastroEventosScreen from './CadastroEventosScreen';

export default class HomeScreen extends Component {

    cadastrarEvento() {
        try {
            this.props.navigation.navigate('CadastroEventosScreen')
        } catch (error) {
            alert(error);
        }
        
    }

    render() {
        return(
            <View style={styles.container}>
                <Text>
                    Home Screen
                </Text>
                <TouchableOpacity 
                    style={styles.botaoCadastro}
                    onPress={() => this.cadastrarEvento()}>
                    <Text style={styles.buttonText}>
                        Cadastrar evento
                    </Text>
                </TouchableOpacity>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        padding: 5
    },
    botaoCadastro: {
        height:45,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom:20,
        width:350,
        borderRadius:30,
        backgroundColor: "limegreen",
    },
    buttonText: {
        color: 'white'
    }
});