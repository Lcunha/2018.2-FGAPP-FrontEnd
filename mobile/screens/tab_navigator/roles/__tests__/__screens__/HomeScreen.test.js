import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import HomeScreen from '../../../roles/screens/HomeScreen';

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', ()=> {
    const tree = renderer.create(<HomeScreen/>).toJSON();
    expect(tree).toMatchSnapshot();
});