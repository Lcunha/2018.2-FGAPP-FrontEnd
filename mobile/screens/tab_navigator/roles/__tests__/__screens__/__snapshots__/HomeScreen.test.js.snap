// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Home Screen
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "limegreen",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 20,
        "opacity": 1,
        "width": 350,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Cadastrar evento
    </Text>
  </View>
</View>
`;
